<a href="https://colab.research.google.com/github/mahatomic/sputnik-filters-research/blob/main/%D0%9F%D0%B5%D1%80%D0%B2%D0%BE%D0%B5_%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_%D0%A2%D0%97_%D0%A1%D0%BF%D1%83%D1%82%D0%BD%D0%B8%D0%BA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

Загружаем датасет

In [ ]:
df = pd.read_csv('/content/events filters к тестовому Product Analyst Q1 2024.csv')

In [ ]:
df

event_category              event_action  \
0       city_landing       price_button_submit   
1       city_landing       price_button_submit   
2       city_landing  search-tools-button_open   
3       city_landing  filters-categories_click   
4       city_landing       price_button_submit   
...              ...                       ...   
19087   city_landing  search-tools-button_open   
19088   city_landing  filters-categories_click   
19089   city_landing  search-tools-button_open   
19090   city_landing       price_button_submit   
19091      City Page                Page Visit   

                                    event_label  total_events  unique_events  
0              Tula / Показать предложения (11)             8              7  
1             Sochi / Показать предложения (39)            12             10  
2                            Penza / Сортировка             2              2  
3      Gelendzhik / ЭКСКУРСИИ В ГЕЛЕНДЖИКЕ ЦЕНЫ             1              1  
4           Moscow / Показать предложения (345)             2              2  
...                                         ...           ...            ...  
19087           Istanbul / ЭКСКУРСИИ В СТАМБУЛЕ            38             32  
19088                          Ufa / Пешеходные             3              3  
19089              Yerevan / Фильтры Выбрано: 0            16             16  
19090         Pskov / Показать предложения (35)             7              7  
19091                                  Auckland            33             25  

[19092 rows x 5 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19092 entries, 0 to 19091
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   event_category  19092 non-null  object
 1   event_action    19092 non-null  object
 2   event_label     19092 non-null  object
 3   total_events    19092 non-null  int64 
 4   unique_events   19092 non-null  int64 
dtypes: int64(2), object(3)
memory usage: 745.9+ KB


Из event_label выделяем города в отдельный столбец city и информацию, куда кликнул пользователь (после /) в столбец info

In [ ]:
df['city'] = df['event_label'].str.split('/').str[0].str.strip()
df['info'] = df['event_label'].str.split('/').str[1].str.strip()
df

event_category              event_action  \
0       city_landing       price_button_submit   
1       city_landing       price_button_submit   
2       city_landing  search-tools-button_open   
3       city_landing  filters-categories_click   
4       city_landing       price_button_submit   
...              ...                       ...   
19087   city_landing  search-tools-button_open   
19088   city_landing  filters-categories_click   
19089   city_landing  search-tools-button_open   
19090   city_landing       price_button_submit   
19091      City Page                Page Visit   

                                    event_label  total_events  unique_events  \
0              Tula / Показать предложения (11)             8              7   
1             Sochi / Показать предложения (39)            12             10   
2                            Penza / Сортировка             2              2   
3      Gelendzhik / ЭКСКУРСИИ В ГЕЛЕНДЖИКЕ ЦЕНЫ             1              1   
4           Moscow / Показать предложения (345)             2              2   
...                                         ...           ...            ...   
19087           Istanbul / ЭКСКУРСИИ В СТАМБУЛЕ            38             32   
19088                          Ufa / Пешеходные             3              3   
19089              Yerevan / Фильтры Выбрано: 0            16             16   
19090         Pskov / Показать предложения (35)             7              7   
19091                                  Auckland            33             25   

             city                         info  
0            Tula    Показать предложения (11)  
1           Sochi    Показать предложения (39)  
2           Penza                   Сортировка  
3      Gelendzhik  ЭКСКУРСИИ В ГЕЛЕНДЖИКЕ ЦЕНЫ  
4          Moscow   Показать предложения (345)  
...           ...                          ...  
19087    Istanbul         ЭКСКУРСИИ В СТАМБУЛЕ  
19088         Ufa                   Пешеходные  
19089     Yerevan           Фильтры Выбрано: 0  
19090       Pskov    Показать предложения (35)  
19091    Auckland                          NaN  

[19092 rows x 7 columns]

Группируем по действиям с подсчетом общего количества всех и уникальных событий

In [ ]:
grouped_on_action_df = df.groupby('event_action')[['total_events', 'unique_events']].sum().reset_index()
grouped_on_action_df

event_action  total_events  unique_events
0                 Page Visit       2124993        1384478
1        clear_filter_mobile          3599           2788
2        dates_filter_mobile         13038          10272
3             end_date_click         14638          11738
4   filters-categories_click         80065          62539
5          pay-type_checkbox          7321           5465
6         price_button_close         14247          11646
7        price_button_submit         71890          63063
8                price_first          8572           6676
9               price_second          2826           1995
10               price_third           904            595
11  search-tools-button_open         94210          85773
12          start_date_click         19939          14501
13      ticket-type_checkbox         22882          17918

Общее количество событий по действию search-tools-button_open больше, чем
по действиям filters-categories_click	и price_button_submit, которые указывают на применение фильтра

In [ ]:
search_df = df[df['event_action'] == 'search-tools-button_open']
search_df.head(100)

event_category              event_action  \
2     city_landing  search-tools-button_open   
13    city_landing  search-tools-button_open   
18    city_landing  search-tools-button_open   
22    city_landing  search-tools-button_open   
27    city_landing  search-tools-button_open   
..             ...                       ...   
399   city_landing  search-tools-button_open   
409   city_landing  search-tools-button_open   
411   city_landing  search-tools-button_open   
412   city_landing  search-tools-button_open   
415   city_landing  search-tools-button_open   

                                           event_label  total_events  \
2                                   Penza / Сортировка             2   
13                          Crete / Фильтры Выбрано: 1             9   
18                                      Irkutsk / КБЖД             2   
22                    Adler / КАКИЕ ЭКСКУРСИИ В АДЛЕРЕ             1   
27                               Haifa / Категории Все             4   
..                                                 ...           ...   
399                   Dubai / Категории Индивидуальные             1   
409                          Petrozavodsk / Выбрано: 3             2   
411  Tbilisi / ИНДИВИДУАЛЬНЫЕ ЭКСКУРСИИ В ТБИЛИСИ Н...             2   
412  Tbilisi / ИНДИВИДУАЛЬНЫЕ ЭКСКУРСИИ ПО ГРУЗИИ И...             2   
415                                 Dubai / Выбрано: 3            18   

     unique_events          city  \
2                2         Penza   
13               9         Crete   
18               2       Irkutsk   
22               1         Adler   
27               4         Haifa   
..             ...           ...   
399              1         Dubai   
409              2  Petrozavodsk   
411              2       Tbilisi   
412              1       Tbilisi   
415             15         Dubai   

                                                  info  
2                                           Сортировка  
13                                  Фильтры Выбрано: 1  
18                                                КБЖД  
22                            КАКИЕ ЭКСКУРСИИ В АДЛЕРЕ  
27                                       Категории Все  
..                                                 ...  
399                           Категории Индивидуальные  
409                                         Выбрано: 3  
411  ИНДИВИДУАЛЬНЫЕ ЭКСКУРСИИ В ТБИЛИСИ НА РУССКОМ ...  
412      ИНДИВИДУАЛЬНЫЕ ЭКСКУРСИИ ПО ГРУЗИИ ИЗ ТБИЛИСИ  
415                                         Выбрано: 3  

[100 rows x 7 columns]

Фактически действие search-tools-button_open - это просто открытие фильтра, но не обязательно его использование в контексте применения, поэтому в расчете это действие учитывать не будем

In [ ]:
filtered_df = df[df['event_action'] == 'filters-categories_click'].copy()
filtered_df.head(100)

event_category              event_action  \
3     city_landing  filters-categories_click   
9     city_landing  filters-categories_click   
11    city_landing  filters-categories_click   
12    city_landing  filters-categories_click   
15    city_landing  filters-categories_click   
..             ...                       ...   
555   city_landing  filters-categories_click   
564   city_landing  filters-categories_click   
572   city_landing  filters-categories_click   
578   city_landing  filters-categories_click   
583   city_landing  filters-categories_click   

                                           event_label  total_events  \
3             Gelendzhik / ЭКСКУРСИИ В ГЕЛЕНДЖИКЕ ЦЕНЫ             1   
9                                   Krakow / Необычные             3   
11                   Barcelona / Сначала самые дешевые            62   
12                           Novorossiysk / Автобусные            42   
15                           Sergiev Posad / Групповые            10   
..                                                 ...           ...   
555                                  Cairo / Групповые            20   
564                      Gorno-Altaysk / Автомобильные             1   
572                                   Minsk / Выездные            20   
578  Sortavala / ШХЕРЫ ЛАДОЖСКОГО ОЗЕРА ЭКСКУРСИЯ И...             1   
583                 Sharm-el-Sheikh / Гастрономические            10   

     unique_events             city  \
3                1       Gelendzhik   
9                3           Krakow   
11              50        Barcelona   
12              32     Novorossiysk   
15               8    Sergiev Posad   
..             ...              ...   
555             17            Cairo   
564              1    Gorno-Altaysk   
572             17            Minsk   
578              1        Sortavala   
583              5  Sharm-el-Sheikh   

                                              info  
3                      ЭКСКУРСИИ В ГЕЛЕНДЖИКЕ ЦЕНЫ  
9                                        Необычные  
11                           Сначала самые дешевые  
12                                      Автобусные  
15                                       Групповые  
..                                             ...  
555                                      Групповые  
564                                  Автомобильные  
572                                       Выездные  
578  ШХЕРЫ ЛАДОЖСКОГО ОЗЕРА ЭКСКУРСИЯ ИЗ СОРТАВАЛА  
583                               Гастрономические  

[100 rows x 7 columns]

В действии filters-categories_click отражено использование пользователем сортировок и категорий в рамках фильтрации

Распределим использование категорий и сортировок по разным датафреймам и удалим из расчета данные, обозначенные капсом, так как предполагаю, что это грязные данные, которые не имеют отношения к фильтрации

Удаляем строчки, где встречается текст капсом

In [ ]:
filtered_df = filtered_df[~filtered_df['info'].str.isupper()]
filtered_df

event_category              event_action  \
9       city_landing  filters-categories_click   
11      city_landing  filters-categories_click   
12      city_landing  filters-categories_click   
15      city_landing  filters-categories_click   
26      city_landing  filters-categories_click   
...              ...                       ...   
19060   city_landing  filters-categories_click   
19069   city_landing  filters-categories_click   
19074   city_landing  filters-categories_click   
19075   city_landing  filters-categories_click   
19088   city_landing  filters-categories_click   

                               event_label  total_events  unique_events  \
9                       Krakow / Необычные             3              3   
11       Barcelona / Сначала самые дешевые            62             50   
12               Novorossiysk / Автобусные            42             32   
15               Sergiev Posad / Групповые            10              8   
26                         Tula / Обзорные            23             17   
...                                    ...           ...            ...   
19060      Omsk / Сначала самые популярные             3              3   
19069  Taganrog / Сначала самые популярные             2              2   
19074                    Belek / Для детей            21             15   
19075                     Rome / На сигвее             1              1   
19088                     Ufa / Пешеходные             3              3   

                city                      info  
9             Krakow                 Необычные  
11         Barcelona     Сначала самые дешевые  
12      Novorossiysk                Автобусные  
15     Sergiev Posad                 Групповые  
26              Tula                  Обзорные  
...              ...                       ...  
19060           Omsk  Сначала самые популярные  
19069       Taganrog  Сначала самые популярные  
19074          Belek                 Для детей  
19075           Rome                 На сигвее  
19088            Ufa                Пешеходные  

[2886 rows x 7 columns]

Выделим сначала в отдельный датафрейм события использования сортировок и подсчитаем их общее количество

In [ ]:
phrases = [
    'Сначала самые дешевые',
    'Сначала самые дорогие',
    'Сначала самые популярные',
    'По рейтингу'
]
or_phrases = '|'.join(phrases)
sort_df = filtered_df[filtered_df['info'].str.contains(or_phrases, case=False, na=False)].copy()
sort_df

event_category              event_action  \
11      city_landing  filters-categories_click   
130     city_landing  filters-categories_click   
193     city_landing  filters-categories_click   
266     city_landing  filters-categories_click   
288     city_landing  filters-categories_click   
...              ...                       ...   
19011   city_landing  filters-categories_click   
19034   city_landing  filters-categories_click   
19050   city_landing  filters-categories_click   
19060   city_landing  filters-categories_click   
19069   city_landing  filters-categories_click   

                                       event_label  total_events  \
11               Barcelona / Сначала самые дешевые            62   
130                         Florence / По рейтингу             2   
193             Bryansk / Сначала самые популярные             1   
266                 Grozny / Сначала самые дешевые            22   
288           Thessaloniki / Сначала самые дешевые             3   
...                                            ...           ...   
19011  Saint Petersburg / Сначала самые популярные           798   
19034   Sharm-el-Sheikh / Сначала самые популярные           245   
19050                       Belgrade / По рейтингу             5   
19060              Omsk / Сначала самые популярные             3   
19069          Taganrog / Сначала самые популярные             2   

       unique_events              city                      info  
11                50         Barcelona     Сначала самые дешевые  
130                2          Florence               По рейтингу  
193                1           Bryansk  Сначала самые популярные  
266               20            Grozny     Сначала самые дешевые  
288                3      Thessaloniki     Сначала самые дешевые  
...              ...               ...                       ...  
19011            482  Saint Petersburg  Сначала самые популярные  
19034            224   Sharm-el-Sheikh  Сначала самые популярные  
19050              5          Belgrade               По рейтингу  
19060              3              Omsk  Сначала самые популярные  
19069              2          Taganrog  Сначала самые популярные  

[684 rows x 7 columns]

In [ ]:
total_sort = sort_df[['total_events', 'unique_events']].sum()
res_sort = total_sort.to_frame('total_sum')
res_sort

total_sum
total_events       20430
unique_events      17382

Получаем, что сортировка была использована 20430 раз, из которых 17382 раз уникальными пользователями

Для создания отдельного датафрейма с категориями удалим из filtered_df строки, которые встречаются в total_sort

In [ ]:
cat_df = filtered_df[~filtered_df.index.isin(sort_df.index)]
cat_df

event_category              event_action                event_label  \
9       city_landing  filters-categories_click         Krakow / Необычные   
12      city_landing  filters-categories_click  Novorossiysk / Автобусные   
15      city_landing  filters-categories_click  Sergiev Posad / Групповые   
26      city_landing  filters-categories_click            Tula / Обзорные   
46      city_landing  filters-categories_click   Novosibirsk / Популярные   
...              ...                       ...                        ...   
19056   city_landing  filters-categories_click        Vologda / Групповые   
19057   city_landing  filters-categories_click        Hurghada / Пирамиды   
19074   city_landing  filters-categories_click          Belek / Для детей   
19075   city_landing  filters-categories_click           Rome / На сигвее   
19088   city_landing  filters-categories_click           Ufa / Пешеходные   

       total_events  unique_events           city        info  
9                 3              3         Krakow   Необычные  
12               42             32   Novorossiysk  Автобусные  
15               10              8  Sergiev Posad   Групповые  
26               23             17           Tula    Обзорные  
46               33             24    Novosibirsk  Популярные  
...             ...            ...            ...         ...  
19056            35             27        Vologda   Групповые  
19057            71             53       Hurghada    Пирамиды  
19074            21             15          Belek   Для детей  
19075             1              1           Rome   На сигвее  
19088             3              3            Ufa  Пешеходные  

[2202 rows x 7 columns]

In [ ]:
total_cat = cat_df[['total_events', 'unique_events']].sum()
res_cat = total_cat.to_frame('total_sum')
res_cat

total_sum
total_events       57835
unique_events      43863

Таким образом, категории были выбраны 57835 раз, из которых 43863 раз уникальными пользователями

Далее рассмотрим применение фильтров. В расчете не будем учитывать действия clear_filter_mobile	и price_button_close, так как эти действия скорее относятся не к фильтрации, а к ее завершению или прерыванию

Судя по особенностям работы сайта, фильтр применяется сразу при нажатии на вариант из предложенного списка на чекбоксе, и при этом не обязательно нажимать "Показать"

Поэтому в расчете также не будем учитывать действие price_button_submit, но при этом просуммируем количество событий всех возможных вариантов применения фильтров

In [ ]:
filter_actions = [
    'dates_filter_mobile',
    'end_date_click',
    'pay-type_checkbox',
    'price_first',
    'price_second',
    'price_third',
    'start_date_click',
    'ticket-type_checkbox'
]
filt_df = df[df['event_action'].isin(filter_actions)]
filt_df

event_category          event_action  \
6       city_landing      start_date_click   
7       city_landing  ticket-type_checkbox   
16      city_landing   dates_filter_mobile   
21      city_landing           price_third   
34      city_landing        end_date_click   
...              ...                   ...   
19072   city_landing        end_date_click   
19081   city_landing  ticket-type_checkbox   
19084   city_landing     pay-type_checkbox   
19085   city_landing  ticket-type_checkbox   
19086   city_landing  ticket-type_checkbox   

                                             event_label  total_events  \
6                                                Vilnius             9   
7                                  Irkutsk / Мини-группа            29   
16                                     Ivanovo / Сегодня             2   
21                                          Petrozavodsk             6   
34                                                Odessa             1   
...                                                  ...           ...   
19072                                             Almaty             8   
19081  Hurghada / Индивидуальная (гид проведёт экскур...            63   
19084         Alushta / Оплата на месте (без предоплаты)            63   
19085                          Gorokhovets / Мини-группа             1   
19086                               Smolensk / Групповая            22   

       unique_events          city  \
6                  4       Vilnius   
7                 26       Irkutsk   
16                 2       Ivanovo   
21                 4  Petrozavodsk   
34                 1        Odessa   
...              ...           ...   
19072              8        Almaty   
19081             53      Hurghada   
19084             52       Alushta   
19085              1   Gorokhovets   
19086             17      Smolensk   

                                                    info  
6                                                    NaN  
7                                            Мини-группа  
16                                               Сегодня  
21                                                   NaN  
34                                                   NaN  
...                                                  ...  
19072                                                NaN  
19081  Индивидуальная (гид проведёт экскурсию только ...  
19084                   Оплата на месте (без предоплаты)  
19085                                        Мини-группа  
19086                                          Групповая  

[2445 rows x 7 columns]

In [ ]:
total_filt = filt_df[['total_events', 'unique_events']].sum()
res_filt = total_filt.to_frame('total_sum')
res_filt

total_sum
total_events       90120
unique_events      69160

Видим, что общее количество событий использования фильтрации 90120, при этом от уникальных пользователей - 69160

Теперь расчитаем сколько людей пользуются фильтрами и оценим, какие разделы фильтров наиболее востребованы

In [ ]:
total_all = pd.DataFrame({
    'Фильтры': total_filt,
    'Категории': total_cat,
    'Сортировки': total_sort
}).T
total_all.loc['Всего'] = total_all[['total_events', 'unique_events']].agg('sum')
total_all

total_events  unique_events
Фильтры            90120          69160
Категории          57835          43863
Сортировки         20430          17382
Всего             168385         130405

In [ ]:
total_unique_events = total_all.loc['Всего', 'unique_events']
unique_page_visits = grouped_on_action_df[grouped_on_action_df['event_action'] == 'Page Visit']['unique_events']
uniq_total = {
    'total_unique_events': total_unique_events,
    'share_of_uniq_page_visits (%)': (total_unique_events / unique_page_visits * 100). round(2),
}
unique__df = pd.DataFrame(uniq_total)
unique__df

total_unique_events  share_of_uniq_page_visits (%)
0               130405                           9.42

Можно сделать вывод, что 130405 уникальных пользователей пользуются фильтрами всех категорий, что составляет 9,42% от количества уникальных посещений сайта

Рассмотрим распределение уникальных пользователей между разделами фильтров

In [ ]:
unique_data = total_all['unique_events'][:-1]
percentages = (unique_data / unique_data.sum() * 100).round(2)
unique_table = pd.DataFrame({
    'unique_events': unique_data,
    'share (%)': percentages
})
unique_table

unique_events  share (%)
Фильтры             69160      53.03
Категории           43863      33.64
Сортировки          17382      13.33

Наиболее востребован раздел "фильтры" (53,03%), далее идут "категории" (33,03%), и наименее популярны "сортировки" (13,33%)

Далее посмотрим, как пользуются фильтрами по городам

In [ ]:
all_filrters = pd.concat([
    filt_df,
    cat_df,
    sort_df
])
city_group = all_filrters.groupby('city', as_index=False)[['total_events', 'unique_events']].sum()
city_group

city  total_events  unique_events
0          Abakan            20             17
1        Abkhazia             7              7
2     Abrau-Durso           108             96
3       Abu Dhabi            75             54
4           Adler          3022           2353
..            ...           ...            ...
369   bogolyubovo             2              2
370       podolsk             8              5
371  service-city             2              2
372      yantarny            10              8
373       zaraysk            18             16

[374 rows x 3 columns]

In [ ]:
sorted_city_group = city_group.sort_values(['total_events', 'unique_events'], ascending=[False, False])
sorted_city_group

city  total_events  unique_events
135       Kaliningrad         15982          11725
271  Saint Petersburg         15767          11070
203            Moscow          9648           7383
295             Sochi          7500           5876
126          Istanbul          6836           5308
..                ...           ...            ...
293         Singapore             1              1
306            Surgut             1              1
327             Tunis             1              1
342           Vicenza             1              1
355         Wiesbaden             1              1

[374 rows x 3 columns]

В результате получаем, что в топ 5 городов, где используются фильтры, входят Калининград, Санкт-Петербург, Москва, Сочи и Стамбул

In [ ]:
sorted_city_group.tail(50)

city  total_events  unique_events
309        Sviyazhsk             3              3
11            Amiens             3              2
37          Belgorod             3              2
7        Aleksandrov             2              2
32              Bari             2              2
96         Frankfurt             2              2
112        Halkidiki             2              2
155            Krabi             2              2
167          Leipzig             2              2
174    Lloret de Mar             2              2
188          Marburg             2              2
210          Myshkin             2              2
228  Orekhovo-Zuyevo             2              2
242         Petrovac             2              2
250            Poreс             2              2
256             Pula             2              2
276            Samui             2              2
290          Sigulda             2              2
298          Sovetsk             2              2
321           Toledo             2              2
322         Tolyatti             2              2
345          Vitebsk             2              2
369      bogolyubovo             2              2
371     service-city             2              2
193         Medellin             2              1
15           Antwerp             1              1
41         Bethlehem             1              1
43            Bilbao             1              1
46           Bourgas             1              1
60          Cambodia             1              1
79             Diani             1              1
80           Dmitrov             1              1
87         Edinburgh             1              1
115         Hannover             1              1
143          Kazbegi             1              1
165        Las Vegas             1              1
187         Marbella             1              1
191        Marseille             1              1
209           Muscat             1              1
214          Nesvizh             1              1
229         Orenburg             1              1
234            Parma             1              1
235         Partenit             1              1
259   Ras Al Khaimah             1              1
279          Saransk             1              1
293        Singapore             1              1
306           Surgut             1              1
327            Tunis             1              1
342          Vicenza             1              1
355        Wiesbaden             1              1

Городов, где пользуются фильтрами меньше всего, достаточно много, при этом очень много единичного использования. К примеру, такими городами являются Саранск, Сингапур, Сургут, Тунис, Виченца, Висбаден

Рассмотрим, как часто люди пользуются фильтром выбора цены

In [ ]:
price_filters = ['price_first', 'price_second', 'price_third']
price_df = grouped_on_action_df[grouped_on_action_df['event_action'].isin(price_filters)].copy()
price_df

event_action  total_events  unique_events
8    price_first          8572           6676
9   price_second          2826           1995
10   price_third           904            595

Оставим только данные для уникальных пользователей

In [ ]:
price_df = price_df.drop(columns=['total_events'])
price_df

event_action  unique_events
8    price_first           6676
9   price_second           1995
10   price_third            595

Посчитаем общее количество использований фильтров выбора цены

In [ ]:
price_df = price_df.reset_index(drop=True)
price_unique_events = price_df['unique_events'].sum()
price_df.loc[len(price_df)] = ['Всего', price_unique_events]
price_df

event_action  unique_events
0   price_first           6676
1  price_second           1995
2   price_third            595
3         Всего           9266

In [ ]:
price_df['share (%)'] = ((price_df['unique_events'] / price_unique_events) * 100).round(2)
price_df[:-1]

event_action  unique_events  share (%)
0   price_first           6676      72.05
1  price_second           1995      21.53
2   price_third            595       6.42

Больше всего пользователи пользуются единичным применением фильтра цены (72,05%) и меньше всего трехкратным (6,42%), что вполне логично

In [ ]:
total_filter_events = total_all.loc['Фильтры', 'unique_events']
data_uniq_price = {
    'price_unique_events': [price_unique_events],
    'share_unique_filt (%)': [((price_unique_events / total_filter_events) * 100). round(2)],
    'share_unique_filt_all (%)': [((price_unique_events / total_unique_events) * 100).round(2)]
}
unique_price_df = pd.DataFrame(data_uniq_price)
unique_price_df

price_unique_events  share_unique_filt (%)  share_unique_filt_all (%)
0                 9266                   13.4                       7.11

Итак, фильтр выбора цены использовался пользователями 9266 раз, что составляет 13,4% от количества событий использования раздела "фильтры" и 7,11% от количества событий использования в принципе фильтров (всех разделов)

Наконец, любопытно посмотреть, какие фичи из раздела "фильтры" используются пользователями чаще всего, а какие меньше всего

In [ ]:
features_df = grouped_on_action_df[grouped_on_action_df['event_action'].isin(filter_actions)]
features_df

event_action  total_events  unique_events
2    dates_filter_mobile         13038          10272
3         end_date_click         14638          11738
5      pay-type_checkbox          7321           5465
8            price_first          8572           6676
9           price_second          2826           1995
10           price_third           904            595
12      start_date_click         19939          14501
13  ticket-type_checkbox         22882          17918

In [ ]:
features_df = features_df.drop(columns=['total_events']).reset_index(drop=True)

Рассчитаем долю использования фичей уникальными пользователями относительно общего количества использования раздела "фильтры" и расположим их по убыванию

In [ ]:
features_df['share_of_filters (%)'] = (features_df['unique_events'] / total_filter_events * 100).round(2)
features_df = features_df.sort_values('share_of_filters (%)', ascending=False)
features_df

event_action  unique_events  share_of_filters (%)
7  ticket-type_checkbox          17918                 25.91
6      start_date_click          14501                 20.97
1        end_date_click          11738                 16.97
0   dates_filter_mobile          10272                 14.85
3           price_first           6676                  9.65
2     pay-type_checkbox           5465                  7.90
4          price_second           1995                  2.88
5           price_third            595                  0.86

Из расчетов видно, что в разделе "фильтры" чаще всего уникальные пользователи пользуются фильтром на размер группы (25,91%), а далее идут фильтры по датам (начало, конец и сегодня/завтра). Меньше всего пользователи задействуют фильтры на повторное изменение стоимости